bambi/results.py

```diff
@@ -114,19 +132,96 @@ def plot(self, burn_in=0, names=None, annotate=True, hide_transformed=True,
+            names (list): Optional list of variable names to summarize.
+            hide_transformed (bool): If True (default), do not print
+                summary statistics for internally transformed variables.
+        '''
+
+        # if no 'names' specified, filter out unwanted variables
+        if names is None:
+            names = self.untransformed_vars if hide_transformed else self.trace.varnames
+            if exclude_ranefs:
+                names = [x for x in names
+                    if x[2:] not in list(self.model.random_terms.keys())]
+
+        # get the basic DataFrame
+        df = pm.df_summary(self.trace[burn_in:], varnames=names, **kwargs)
+
+        # replace the "__\d" suffixes with an informative factor level name
+        match = [re.match('^(.*)(?:__)(\d+)?$', x) for x in df.index]
+        def replace_with_name(match):
+            term = self.model.terms[match.group(1)[2:]]
+            # handle fixed effects
+            if term in self.model.fixed_terms.values():
+                return term.levels[int(match.group(2))]
+            # handle random effects
+            else:
+                return '{}[{}]'.format(term.name, term.levels[int(match.group(2))])
+        new = [replace_with_name(x) if x is not None else df.index[i]
+            for i,x in enumerate(match)]
+        df.set_index([new], inplace=True)
+
+        # For binomial models with n_trials = 1 (most common use case),
+        # tell user which event is being modeled
+        if self.model.family.name=='binomial' and np.max(self.model.y.data) < 1.01:
+            event = next(i for i,x in enumerate(self.model.y.data.flatten()) if x==1)
+            print('NOTE: Modeling the probability that {}==\'{}\''.format(
+                self.model.y.name, str(self.model.data[self.model.y.name][event])))
+
+        return df
+
+    def get_trace(self, burn_in=0, names=None, hide_transformed=True):
+        '''
+        Returns the MCMC samples in a nice, neat DataFrame.
+        Args:
+            burn_in (int): Number of initial samples to exclude from
+                each chain before returning the trace DataFrame.
+            names (list): Optional list of variable names to get samples for.
+            hide_transformed (bool): If True (default), do not return
+            samples for internally transformed variables.
         '''
         if names is None:
             names = self.untransformed_vars if hide_transformed else self.trace.varnames
 
-        return pm.summary(self.trace[burn_in:], varnames=names, **kwargs)
+        # helper function to label the trace DataFrame columns appropriately
+        def get_cols(var):
+            # handle terms with a single level
+            if len(self.trace[var].shape)==1 or self.trace[var].shape[1]==1:
+                return [var]
+            else:
+                # handle fixed terms with multiple levels
+                # (slice off the 'b_' or 'u_')
```

Comment:
This is unrelated to the current PR, but we may want to consider a package-level config module that lets the user specify whether they want the `b_` and `u_` prefixes added in the first place (we should probably default to not mucking with the names at all).